source/fuzz/pass_recommender_standard.cpp

```diff
@@ -0,0 +1,280 @@
+    // - Inlining the function for the region that was outlined might also be
+    //   fruitful; it will be inlined in a different form
+    return RandomOrderAndNonNull(
+        {pass_instances_->GetAddFunctionCalls(), pass_instances_->GetInlineFunctions()});
+  }
+  if (&pass == pass_instances_->GetPermuteBlocks()) {
+    // No obvious follow-on passes.
+    return {};
+  }
+  if (&pass == pass_instances_->GetPermuteFunctionParameters()) {
+    // No obvious follow-on passes.
+    return {};
+  }
+  if (&pass == pass_instances_->GetPermuteInstructions()) {
+    // No obvious follow-on passes.
+    return {};
+  }
+  if (&pass == pass_instances_->GetPropagateInstructionsUp()) {
+    // No obvious follow-on passes.
+    return {};
+  }
+  if (&pass == pass_instances_->GetPushIdsThroughVariables()) {
+    // - This pass creates synonyms, so it is worth applying them.
+    return RandomOrderAndNonNull({pass_instances_->GetApplyIdSynonyms()});
+  }
+  if (&pass ==
+      pass_instances_->GetReplaceAddsSubsMulsWithCarryingExtended()) {
+    // No obvious follow-on passes.
+    return {};
+  }
+  if (&pass ==
+      pass_instances_->GetReplaceCopyMemoriesWithLoadsStores()) {
+    // No obvious follow-on passes.
+    return {};
+  }
+  if (&pass == pass_instances_->GetReplaceCopyObjectsWithStoresLoads()) {
+    // - We may end up with load/store pairs that could be used to create memory
+    //   copies.
+    return RandomOrderAndNonNull(
+        {pass_instances_->GetReplaceLoadsStoresWithCopyMemories()});
+  }
+  if (&pass ==
+      pass_instances_->GetReplaceLoadsStoresWithCopyMemories()) {
+    // No obvious follow-on passes.
+    return {};
+  }
+  if (&pass == pass_instances_->GetReplaceParameterWithGlobal()) {
+    // No obvious follow-on passes.
+    return {};
+  }
+  if (&pass == pass_instances_->GetReplaceLinearAlgebraInstructions()) {
+    // No obvious follow-on passes.
+    return {};
+  }
+  if (&pass == pass_instances_->GetReplaceParamsWithStruct()) {
+    // TODO: possibly some passes related to composites
+    return {};
+  }
+  if (&pass == pass_instances_->GetSplitBlocks()) {
+    // - More blocks means more chances for adding dead breaks/continues, and
+    //   for adding dead blocks
+    return RandomOrderAndNonNull(
+        {pass_instances_->GetAddDeadBreaks(), pass_instances_->GetAddDeadContinues(),
+         pass_instances_->GetAddDeadBlocks()});
+  }
+  if (&pass == pass_instances_->GetSwapBranchConditionalOperands()) {
+    // No obvious follow-on passes.
+    return {};
+  }
+  assert(false && "Unreachable: every fuzzer pass should be dealt with.");
+
+
+
+}
+
+std::vector<FuzzerPass*> PassRecommenderStandard::RandomOrderAndNonNull(
+    const std::vector<FuzzerPass*>& passes) {
+  std::vector<uint32_t> indices(passes.size());
+  std::iota(indices.begin(), indices.end(), 0);
+  std::vector<FuzzerPass*> result;
+  // TODO: don't necessarily always add the recommendations even if the struct
+  //  fields are non-null.
```

Comment:
One possible solution is to shuffle `indices` and select first `n` of them where `n` is a random number from `[0; indices.size())`.

Reply:
I did this a slightly different way.